gateway/src/main/kotlin/com/makarytskyi/gateway/handler/GlobalExceptionHandler.kt

```diff
@@ -0,0 +1,30 @@
+package com.makarytskyi.gateway.handler
+
+import com.makarytskyi.core.exception.NotFoundException
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.ExceptionHandler
+import org.springframework.web.bind.annotation.RestControllerAdvice
+
+@RestControllerAdvice
+internal class GlobalExceptionHandler {
+    @ExceptionHandler(NotFoundException::class)
+    internal fun handleResourceNotFoundException(): ResponseEntity<ErrorResponse> {
+        val errorResponse = ErrorResponse(
+            HttpStatus.NOT_FOUND.value(),
+        )
+        return ResponseEntity.status(HttpStatus.NOT_FOUND).body(errorResponse)
+    }
+
+    @ExceptionHandler(IllegalArgumentException::class)
+    internal fun handleIllegalArgumentException(): ResponseEntity<ErrorResponse> {
+        val errorResponse = ErrorResponse(
+            HttpStatus.BAD_REQUEST.value(),
+        )
+        return ResponseEntity(errorResponse, HttpStatus.BAD_REQUEST)
```

Comment:
Be consistent.
You have here:
```
return ResponseEntity(errorResponse, HttpStatus.BAD_REQUEST)
```

But in the function above:
```
return ResponseEntity.status(HttpStatus.NOT_FOUND).body(errorResponse)
```

Why the code is different? Am I missing something?

Reply:
No, I will fix it.